### DISCUSSION TITLE
JS Hook to catch 404?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5780] [Q&A] [answered]
I'd like to add some JS middleware that catches 404's before `{"code":404,"message":"Not Found.","data":{}}` is shown to the user. Is this possible?

**Comment by ganigeorgiev** [comment, marked as answer]
Assuming that you are asking about v0.23.0-rc, it depends on what is the end goal but inspecting the `e.Next()` error from any global middleware should work. With goja the caveat is that Go errors are wrapped in `GoError` object and the original one ([ApiError](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.23.0-rc8/tools/router#ApiError)) is under the `value` prop. Or in other words, something like below should work:

```js
routerUse((e) => {
    try {
        e.next()
    } catch (err) {
        if (err.value?.status == 404) {
            // do something...
        }

        throw err // rethrow
    }
})

```

**Reply by benallfree** [reply]
Thank you. I have the same question for v0.22.x and below.

**Reply by gwashark** [reply]
This helps me with making full sites with custom html pages. Thanks!
